python/katana/local/entity_type_manager.pyx

```diff
@@ -0,0 +1,110 @@
+
+
+cdef set[string] handle_result_TypeNameSet(Result[set[string]] res) nogil except *:
+    if not res.has_value():
+        with gil:
+            raise_error_code(res.error())
+    return move(res.value())
+
+cdef class EntityType:
+    """
+    A class representing node/edge types.
+    """
+
+    @property
+    def type_id(self):
+        return self._type_id
+
+    def __str__(self):
+        """:rtype str: Name of the corresponding atomic type"""
+        typename_option = self._type_manager.GetAtomicTypeName(self._type_id)
+        return typename_option.value().decode("utf-8")
+
+cdef class AtomicEntityType(EntityType):
+    """
+    A class representing node/edge actomice types.
+    """
+    @property
+    def name(self):
+        return self._name
+
+    def __str__(self):
+        """:rtype str: Name of the corresponding atomic type"""
+        return self._name.decode("utf-8")
+
+cdef class EntityTypeManager:
+
+    @staticmethod
+    cdef EntityTypeManager make(const CEntityTypeManager *manager):
+        m = <EntityTypeManager>EntityTypeManager.__new__(EntityTypeManager)
+        m.underlying_entity_type_manager = manager
+        return m
+
+    def _make_entity_type(self, type_id):
+        t = EntityType()
+        t._type_manager = self.underlying_entity_type_manager
+        t._type_id = type_id
+        return t
+
+    def _make_atomic_entity_type(self, name):
+        t = AtomicEntityType()
+        t._type_manager = self.underlying_entity_type_manager
+        t._name = name
+        if not self.underlying_entity_type_manager.HasAtomicType(name):
+            raise LookupError(f"{name} does not represent an atomic type")
+        t._type_id = self.underlying_entity_type_manager.GetEntityTypeID(name)
+        return t
+
+    @property
+    def atomic_types(self):
+        """
+        :return: a mapping from atomic type names to enity types.
+        :rtype: dict[str, AtomicEntityType]
+        """
+        atomic_type_names = self.underlying_entity_type_manager.ListAtomicTypes()
+        return {name: self._make_atomic_entity_type(name) for name in atomic_type_names}
+
+    def entity_type_to_type_name_set(self, entity_type):
+        """
+        :param entity_type: Can be a positive integer or an EntityType object
+        :return: a set of strings representing the names of the types for the entity type
+        """
+        if isinstance(entity_type, int):
+            entity_type_id = entity_type
+        elif isinstance(entity_type, EntityType):
+            entity_type_id = entity_type.type_id
+        else:
+            raise ValueError(f"{entity_type}'s type is not supported")
+        if not self.underlying_entity_type_manager.HasEntityType(entity_type_id):
+            raise LookupError(f"{entity_type_id} does not represent a type")
+        return handle_result_TypeNameSet(self.underlying_entity_type_manager.EntityTypeToTypeNameSet(entity_type_id))
```

Comment:
This method doesn't match the mappings we are using for the other parts of the interface. I think in the interface I proposed the the idea was to have a list of all entitytypes and then a function to get all atomic super types (which is the set of names you are returning here). Also this shouldn't return names, but should return `EntityType` objects.

Reply:
Makes sense. Since this PR doesn't touch `DynamicBitsetSlow`. I'll remove this function.

Reply:
Sounds good. Re can come back to it in the next PR.